# æ¼”ç¿’A

ä»¥ä¸‹ã‚’ JupyterHub ã§å®Ÿéš›ã«å®Ÿè¡Œã—ã¾ã—ã‚‡ã†ã€‚

- [æ¼”ç¿’A](#æ¼”ç¿’a)
  - [1. ãƒ‡ãƒ¼ã‚¿ã®ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰](#1-ãƒ‡ãƒ¼ã‚¿ã®ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰)
  - [2. ä»¥é™ã¯ R ã®ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã‚’ä½œæˆã—ã¦å§‹ã‚ã¾ã—ã‚‡ã†](#2-ä»¥é™ã¯-r-ã®ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã‚’ä½œæˆã—ã¦å§‹ã‚ã¾ã—ã‚‡ã†)
  - [3. `tidyverse` ã«ã¤ã„ã¦å­¦ã¶](#3-tidyverse-ã«ã¤ã„ã¦å­¦ã¶)
    - [`tidyverse` ã¨ã¯](#tidyverse-ã¨ã¯)
    - [tidyverse ã‚’ä½¿ã†ã«ã¯](#tidyverse-ã‚’ä½¿ã†ã«ã¯)
  - [4. ãƒ†ãƒ¼ãƒ–ãƒ«ï¼ˆè¡¨ï¼‰ãƒ‡ãƒ¼ã‚¿ã®èª­ã¿è¾¼ã¿ï¼š `read_tsv()`, `read_csv()`](#4-ãƒ†ãƒ¼ãƒ–ãƒ«è¡¨ãƒ‡ãƒ¼ã‚¿ã®èª­ã¿è¾¼ã¿-read_tsv-read_csv)
  - [5. ãƒ‡ãƒ¼ã‚¿ãƒ†ãƒ¼ãƒ–ãƒ«ï¼ˆè¡¨ï¼‰ã®å½¢ã‚’èª¿ã¹ã‚‹ã®ã«ä¾¿åˆ©ãªé–¢æ•°ï¼š `dim()`, `head()`, `str()`](#5-ãƒ‡ãƒ¼ã‚¿ãƒ†ãƒ¼ãƒ–ãƒ«è¡¨ã®å½¢ã‚’èª¿ã¹ã‚‹ã®ã«ä¾¿åˆ©ãªé–¢æ•°-dim-head-str)
  - [6. ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã‹ã‚‰ç‰¹å®šã®æ¡ä»¶ã«å½“ã¦ã¯ã¾ã‚‹è¡Œã‚’æŠ½å‡ºã™ã‚‹ï¼š `filter()`](#6-ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã‹ã‚‰ç‰¹å®šã®æ¡ä»¶ã«å½“ã¦ã¯ã¾ã‚‹è¡Œã‚’æŠ½å‡ºã™ã‚‹-filter)
  - [7. ç‰¹å®šã®åˆ—ã‚’æŠ½å‡ºã™ã‚‹ï¼š `select()`](#7-ç‰¹å®šã®åˆ—ã‚’æŠ½å‡ºã™ã‚‹-select)
  - [8. ãƒ‘ã‚¤ãƒ—æ¼”ç®—å­ `%>%`](#8-ãƒ‘ã‚¤ãƒ—æ¼”ç®—å­-)
  - [9. åˆ—ã‚’åŠ ãˆã‚‹ï¼š `mutate()`](#9-åˆ—ã‚’åŠ ãˆã‚‹-mutate)
  - [8. ï¼’ã¤ã®åˆ—ã‚’çµåˆã™ã‚‹](#8-ï¼’ã¤ã®åˆ—ã‚’çµåˆã™ã‚‹)
  - [`tumor_communication.ipynb` ã‚’å®Ÿè¡Œã™ã‚‹](#tumor_communicationipynb-ã‚’å®Ÿè¡Œã™ã‚‹)
    - [ã“ã‚Œã¯ãªã‚“ãªã®ã‹ï¼Ÿ](#ã“ã‚Œã¯ãªã‚“ãªã®ã‹)
    - [å®Ÿè¡Œæ–¹æ³•](#å®Ÿè¡Œæ–¹æ³•)
    - [ä½¿ã†ãƒ‡ãƒ¼ã‚¿ã®æ§‹é€ ](#ä½¿ã†ãƒ‡ãƒ¼ã‚¿ã®æ§‹é€ )
  - [åŸºæœ¬èª²é¡Œ](#åŸºæœ¬èª²é¡Œ)
    - [åŸºæœ¬èª²é¡ŒA-1](#åŸºæœ¬èª²é¡Œa-1)
    - [åŸºæœ¬èª²é¡ŒA-2](#åŸºæœ¬èª²é¡Œa-2)
    - [åŸºæœ¬èª²é¡ŒA-3](#åŸºæœ¬èª²é¡Œa-3)
    - [åŸºæœ¬èª²é¡ŒA-4](#åŸºæœ¬èª²é¡Œa-4)
    - [åŸºæœ¬èª²é¡ŒA-5](#åŸºæœ¬èª²é¡Œa-5)

## 1. ãƒ‡ãƒ¼ã‚¿ã®ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰

ã‚¿ãƒ¼ãƒŸãƒŠãƒ«ã‚’é–‹ãã€ä»¥ä¸‹ã®ã‚³ãƒãƒ³ãƒ‰ã‚’å®Ÿè¡Œã—ã¦ãƒ‡ãƒ¼ã‚¿ã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã—ã¦ãã ã•ã„ã€‚

```bash
$ mkdir data

$ cd data

$ wget https://raw.githubusercontent.com/bioinfo-tsukuba/20201226-EB62104-bioinformatics-course/master/tutorial/data/GSE59831_processed_data_FPKM.txt

$ wget https://github.com/bioinfo-tsukuba/20201226-EB62104-bioinformatics-course/raw/master/tutorial/data/CCCExplorer/CD11CB_output/gene_exp.diff

$ wget https://github.com/bioinfo-tsukuba/20201226-EB62104-bioinformatics-course/raw/master/tutorial/data/CCCExplorer/HOM_MouseHumanSequence.rpt

$ wget https://github.com/bioinfo-tsukuba/20201226-EB62104-bioinformatics-course/raw/master/tutorial/data/CCCExplorer/EP_output/genes.read_group_tracking

$ wget https://github.com/bioinfo-tsukuba/20201226-EB62104-bioinformatics-course/raw/master/tutorial/data/CCCExplorer/LR_manual_revised.txt

$ cd ../
```

## 2. ä»¥é™ã¯ R ã®ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã‚’ä½œæˆã—ã¦å§‹ã‚ã¾ã—ã‚‡ã†

- R ã®ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã‚’æ–°ã—ãä½œæˆã—ã¦ãã ã•ã„
- ã‚¿ã‚¤ãƒˆãƒ«ã‚’ "training_A" ã«å¤‰æ›´ã—ã¦ãã ã•ã„

## 3. `tidyverse` ã«ã¤ã„ã¦å­¦ã¶

### `tidyverse` ã¨ã¯

- ãƒ‡ãƒ¼ã‚¿ã‚µã‚¤ã‚¨ãƒ³ã‚¹ã«ä¾¿åˆ©ãªãƒ‘ãƒƒã‚±ãƒ¼ã‚¸ã®é›†åˆä½“
- ä»Šæ—¥ã®æˆæ¥­ã§æ¯å›ä½¿ç”¨ã—ã¾ã™

### tidyverse ã‚’ä½¿ã†ã«ã¯

ãã®ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã®ä¸­ã§ä¸€åº¦ã€ä»¥ä¸‹ã‚’å®Ÿè¡Œã™ã‚‹å¿…è¦ãŒã‚ã‚Šã¾ã™ã€‚

```R
library(tidyverse)
```

ä»¥ä¸‹ã®ã‚ˆã†ãªãƒ¡ãƒƒã‚»ãƒ¼ã‚¸ãŒå‡ºã¦ãã‚‹ã¨æ€ã„ã¾ã™ï¼ˆå•é¡Œã‚ã‚Šã¾ã›ã‚“ï¼‰


    â”€â”€ [1mAttaching packages[22m â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.3.0 â”€â”€
    
    [32mâœ”[39m [34mggplot2[39m 3.3.3     [32mâœ”[39m [34mpurrr  [39m 0.3.4
    [32mâœ”[39m [34mtibble [39m 3.0.5     [32mâœ”[39m [34mdplyr  [39m 1.0.3
    [32mâœ”[39m [34mtidyr  [39m 1.1.2     [32mâœ”[39m [34mstringr[39m 1.4.0
    [32mâœ”[39m [34mreadr  [39m 1.4.0     [32mâœ”[39m [34mforcats[39m 0.5.1
    
    â”€â”€ [1mConflicts[22m â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
    [31mâœ–[39m [34mdplyr[39m::[32mfilter()[39m masks [34mstats[39m::filter()
    [31mâœ–[39m [34mdplyr[39m::[32mlag()[39m    masks [34mstats[39m::lag()
    

## 4. ãƒ†ãƒ¼ãƒ–ãƒ«ï¼ˆè¡¨ï¼‰ãƒ‡ãƒ¼ã‚¿ã®èª­ã¿è¾¼ã¿ï¼š `read_tsv()`, `read_csv()`

- ãƒ‡ãƒ¼ã‚¿ã®èª­ã¿è¾¼ã¿ã«ã¯ `read_tsv()`, `read_csv()` ã‚’ç”¨ã„ã¾ã™
- TSV ã¯ã‚¿ãƒ–åŒºåˆ‡ã‚Šãƒ•ã‚¡ã‚¤ãƒ«ã€CSV ã¯ã‚³ãƒ³ãƒåŒºåˆ‡ã‚Šãƒ•ã‚¡ã‚¤ãƒ«ã®ã“ã¨ã§ã™
  - 1\.ã§ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã—ãŸãƒ•ã‚¡ã‚¤ãƒ«ãŒTSVã‹CSVã‹ã¯ã€ã‚¿ãƒ¼ãƒŸãƒŠãƒ«ã§ `head` ã‚³ãƒãƒ³ãƒ‰ã§ç¢ºã‹ã‚ã‚‰ã‚Œã¾ã™
- ä¾‹ãˆã°ã€ `GSE59831_processed_data_FPKM.txt` ã¯ã‚¿ãƒ–åŒºåˆ‡ã‚Šãƒ•ã‚¡ã‚¤ãƒ«ãªã®ã§ã€ `read_tsv()` ã§èª­ã¿è¾¼ã‚€ã“ã¨ãŒã§ãã¾ã™
  - `col_names=TRUE` ã¯ã€ï¼‘è¡Œç›®ã‚’åˆ—åã¨ã—ã¦æ‰±ã£ã¦ã»ã—ã„ã¨ã„ã†æ„å‘³ã§ã™ã€‚

ä»¥ä¸‹ã®ã‚³ãƒãƒ³ãƒ‰ã¯ã€ `GSE59831_processed_data_FPKM.txt` ã‚’èª­ã¿è¾¼ã¿ã€ `df1` ã¨ã„ã†ã‚ªãƒ–ã‚¸ã‚§ã‚¯ãƒˆã«æ ¼ç´ã™ã‚‹ã“ã¨ã‚’è¡¨ã—ã¾ã™ã€‚

```R
df1 <- read_tsv("data/GSE59831_processed_data_FPKM.txt", col_names=TRUE)
```


## 5. ãƒ‡ãƒ¼ã‚¿ãƒ†ãƒ¼ãƒ–ãƒ«ï¼ˆè¡¨ï¼‰ã®å½¢ã‚’èª¿ã¹ã‚‹ã®ã«ä¾¿åˆ©ãªé–¢æ•°ï¼š `dim()`, `head()`, `str()`

- `head()` ã¯ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã®æœ€åˆã®6è¡Œã‚’è¡¨ç¤ºã—ã¾ã™
- `dim()` ã¯ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã®è¡Œæ•°ã¨åˆ—æ•°ã‚’å‡ºåŠ›ã—ã¾ã™
- `str()` ã¯è¡Œæ•°ã¨åˆ—æ•°ã®ã»ã‹ã€ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ãŒã©ã‚“ãªåˆ—ï¼ˆåˆ—åã€åˆ—ã®ãƒ‡ãƒ¼ã‚¿å‹ï¼‰ã‚’æŒã¤ã‹ã‚’è¡¨ç¤ºã—ã¾ã™
- `nrow()`ã€`ncol()` ã¯ãã‚Œãã‚Œè¡Œæ•°ã€åˆ—æ•°ã‚’å‡ºåŠ›ã—ã¾ã™

ä»¥ä¸‹ã«å®Ÿè¡Œä¾‹ã‚’ç¤ºã—ã¾ã™ã€‚

```R
head(df1)
```


<table class="dataframe">
<caption>A tibble: 6 Ã— 29</caption>
<thead>
	<tr><th scope=col>mice_gene_symbol</th><th scope=col>human_gene_symbol</th><th scope=col>Tum1</th><th scope=col>Tum2</th><th scope=col>Tum3</th><th scope=col>WT1</th><th scope=col>WT2</th><th scope=col>Tum4</th><th scope=col>Tum5</th><th scope=col>WT3</th><th scope=col>â‹¯</th><th scope=col>Tum11</th><th scope=col>WT8</th><th scope=col>WT9</th><th scope=col>WT10</th><th scope=col>Tum12</th><th scope=col>Tum13</th><th scope=col>Tum14</th><th scope=col>WT11</th><th scope=col>WT12</th><th scope=col>WT13</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>â‹¯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>0610007P14Rik</td><td>C14orf1 </td><td> 9.31368</td><td>10.13680</td><td>12.26660</td><td>11.92950</td><td>11.34170</td><td>10.45680</td><td>10.44590</td><td>13.05840</td><td>â‹¯</td><td>21.99580</td><td>33.98430</td><td>23.50140</td><td>24.80460</td><td>17.16570</td><td>12.51170</td><td>16.03630</td><td>14.69160</td><td>14.03290</td><td>14.20440</td></tr>
	<tr><td>0610009D07Rik</td><td>SF3B14  </td><td>25.35280</td><td>32.25990</td><td>26.61000</td><td>29.73440</td><td>37.16520</td><td>37.57330</td><td>30.74340</td><td>32.30680</td><td>â‹¯</td><td>28.76030</td><td>21.51540</td><td>30.83840</td><td>19.26180</td><td>35.14690</td><td>33.53620</td><td>42.73760</td><td>29.27860</td><td>29.37960</td><td>28.06600</td></tr>
	<tr><td>0610009O20Rik</td><td>KIAA0141</td><td>12.45760</td><td>10.17690</td><td>11.95220</td><td>10.62740</td><td> 7.04763</td><td> 9.63765</td><td>11.11900</td><td>13.35890</td><td>â‹¯</td><td>20.57930</td><td>18.12230</td><td>18.65090</td><td>18.58170</td><td>15.60410</td><td>15.34920</td><td>13.04400</td><td>16.76130</td><td>15.91300</td><td>18.05510</td></tr>
	<tr><td>0610010F05Rik</td><td>KIAA1841</td><td> 5.84091</td><td> 6.54488</td><td> 6.65366</td><td> 4.80074</td><td> 6.06263</td><td> 5.57042</td><td> 5.21223</td><td> 5.56449</td><td>â‹¯</td><td> 8.04684</td><td> 5.92199</td><td> 6.23551</td><td> 4.73364</td><td> 5.98708</td><td> 4.26902</td><td> 4.16373</td><td> 3.47511</td><td> 3.71875</td><td> 3.95344</td></tr>
	<tr><td>0610010K14Rik</td><td>C17orf49</td><td>25.63690</td><td>24.08560</td><td>20.17560</td><td>25.34280</td><td>29.14040</td><td>31.46110</td><td>28.30990</td><td>22.17650</td><td>â‹¯</td><td>18.00200</td><td>24.00260</td><td>21.85860</td><td>23.72630</td><td>28.46520</td><td>31.19330</td><td>26.40880</td><td>32.53430</td><td>26.93620</td><td>29.38080</td></tr>
	<tr><td>0610011F06Rik</td><td>C16orf13</td><td> 3.48174</td><td> 5.79169</td><td> 3.73809</td><td> 5.94575</td><td> 5.98899</td><td> 9.04125</td><td> 5.84235</td><td> 7.36357</td><td>â‹¯</td><td>42.61460</td><td>38.91730</td><td>33.56900</td><td>37.34490</td><td>20.33750</td><td>18.81900</td><td>14.17990</td><td>14.93870</td><td>16.12340</td><td>13.39220</td></tr>
</tbody>
</table>




```R
dim(df1)
```


    16024   29

```R
str(df1)
```

    tibble [16,024 Ã— 29] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
     $ mice_gene_symbol : chr [1:16024] "0610007P14Rik" "0610009D07Rik" "0610009O20Rik" "0610010F05Rik" ...
     $ human_gene_symbol: chr [1:16024] "C14orf1" "SF3B14" "KIAA0141" "KIAA1841" ...
     $ Tum1             : num [1:16024] 9.31 25.35 12.46 5.84 25.64 ...
     $ Tum2             : num [1:16024] 10.14 32.26 10.18 6.54 24.09 ...
     $ Tum3             : num [1:16024] 12.27 26.61 11.95 6.65 20.18 ...
     $ WT1              : num [1:16024] 11.9 29.7 10.6 4.8 25.3 ...
     $ WT2              : num [1:16024] 11.34 37.17 7.05 6.06 29.14 ...
     $ Tum4             : num [1:16024] 10.46 37.57 9.64 5.57 31.46 ...
     $ Tum5             : num [1:16024] 10.45 30.74 11.12 5.21 28.31 ...
     $ WT3              : num [1:16024] 13.06 32.31 13.36 5.56 22.18 ...
     $ WT4              : num [1:16024] 8.74 25.08 12.65 3.93 28.72 ...
     $ Tum6             : num [1:16024] 3.9 31.15 3.93 1.56 22.08 ...
     $ Tum7             : num [1:16024] 2.54 33.48 3.74 1.68 19.04 ...
     $ Tum8             : num [1:16024] 3.34 36.3 3.67 1.24 18.23 ...
     $ WT5              : num [1:16024] 1.14 26.84 2.88 1.05 6.45 ...
     $ WT6              : num [1:16024] 1.553 24.426 3.984 0.645 11.169 ...
     $ WT7              : num [1:16024] 2.7 22.99 3.74 1.1 8.6 ...
     $ Tum9             : num [1:16024] 21.51 30.39 19.15 7.35 20.08 ...
     $ Tum10            : num [1:16024] 25.08 23.83 20.82 7.22 22.06 ...
     $ Tum11            : num [1:16024] 22 28.76 20.58 8.05 18 ...
     $ WT8              : num [1:16024] 33.98 21.52 18.12 5.92 24 ...
     $ WT9              : num [1:16024] 23.5 30.84 18.65 6.24 21.86 ...
     $ WT10             : num [1:16024] 24.8 19.26 18.58 4.73 23.73 ...
     $ Tum12            : num [1:16024] 17.17 35.15 15.6 5.99 28.47 ...
     $ Tum13            : num [1:16024] 12.51 33.54 15.35 4.27 31.19 ...
     $ Tum14            : num [1:16024] 16.04 42.74 13.04 4.16 26.41 ...
     $ WT11             : num [1:16024] 14.69 29.28 16.76 3.48 32.53 ...
     $ WT12             : num [1:16024] 14.03 29.38 15.91 3.72 26.94 ...
     $ WT13             : num [1:16024] 14.2 28.07 18.06 3.95 29.38 ...
     - attr(*, "spec")=
      .. cols(
      ..   mice_gene_symbol = [31mcol_character()[39m,
      ..   human_gene_symbol = [31mcol_character()[39m,
      ..   Tum1 = [32mcol_double()[39m,
      ..   Tum2 = [32mcol_double()[39m,
      ..   Tum3 = [32mcol_double()[39m,
      ..   WT1 = [32mcol_double()[39m,
      ..   WT2 = [32mcol_double()[39m,
      ..   Tum4 = [32mcol_double()[39m,
      ..   Tum5 = [32mcol_double()[39m,
      ..   WT3 = [32mcol_double()[39m,
      ..   WT4 = [32mcol_double()[39m,
      ..   Tum6 = [32mcol_double()[39m,
      ..   Tum7 = [32mcol_double()[39m,
      ..   Tum8 = [32mcol_double()[39m,
      ..   WT5 = [32mcol_double()[39m,
      ..   WT6 = [32mcol_double()[39m,
      ..   WT7 = [32mcol_double()[39m,
      ..   Tum9 = [32mcol_double()[39m,
      ..   Tum10 = [32mcol_double()[39m,
      ..   Tum11 = [32mcol_double()[39m,
      ..   WT8 = [32mcol_double()[39m,
      ..   WT9 = [32mcol_double()[39m,
      ..   WT10 = [32mcol_double()[39m,
      ..   Tum12 = [32mcol_double()[39m,
      ..   Tum13 = [32mcol_double()[39m,
      ..   Tum14 = [32mcol_double()[39m,
      ..   WT11 = [32mcol_double()[39m,
      ..   WT12 = [32mcol_double()[39m,
      ..   WT13 = [32mcol_double()[39m
      .. )


## 6. ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã‹ã‚‰ç‰¹å®šã®æ¡ä»¶ã«å½“ã¦ã¯ã¾ã‚‹è¡Œã‚’æŠ½å‡ºã™ã‚‹ï¼š `filter()`

- `filter()` ã¯ç¬¬ï¼‘å¼•æ•°ã«ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã€ç¬¬ï¼’å¼•æ•°ã«æ¡ä»¶å¼ã‚’æ›¸ãã¨ã€æ¡ä»¶ã«å½“ã¦ã¯ã¾ã‚‹è¡Œã‚’æŠ½å‡ºã™ã‚‹

ä»¥ä¸‹ã«ä¾‹ã‚’ç¤ºã™

```R
# Tum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ãã„è¡Œã‚’æŠ½å‡º
filter(df1, Tum1 > 100)

# Tum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ããã€WT1 åˆ—ã®å€¤ãŒ10ã‚ˆã‚Šå°ã•ã„è¡Œã‚’æŠ½å‡º
# (& ã¯åŒæ™‚ã«æº€ãŸã™ã¹ãæ¡ä»¶ã‚’çµ„ã¿åˆã‚ã›ã‚‹)
filter(df1, Tum1 > 100 & WT1 <10)

# Tum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ãã„ã‹ã€Tum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ãã„è¡Œã‚’æŠ½å‡º
# (| ã¯ã„ãšã‚Œã‹æº€ãŸã™ã¹ãæ¡ä»¶ã‚’çµ„ã¿åˆã‚ã›ã‚‹)
filter(df1, Tum1 > 100 | Tum2 > 100)

# ã€ŒTum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ãã„ã‹ã€Tum1 åˆ—ã®å€¤ãŒ100ã‚ˆã‚Šå¤§ãã„ã€
# ã‹ã¤ã€ŒWT1åˆ—ã®å€¤ãŒ10ã‚ˆã‚Šå°ã•ã„ã‹ã€WT2åˆ—ã®å€¤ãŒ10ã‚ˆã‚Šå°ã•ã„ã€è¡Œã‚’æŠ½å‡º
# (ã‚«ãƒƒã‚³ã§å›²ã‚€ã“ã¨ã§ã²ã¨ã¾ã¨ã¾ã‚Šã¨ã—ã¦æ‰±ãˆã‚‹)
filter(df1, (Tum1>100 | Tum2>100) & (WT1<10 | WT2<10) ) 
```

ãªãŠã€æŠ½å‡ºã•ã‚ŒãŸçµæœã¯ã€å…ƒã®ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ è‡ªä½“ã«ã¯ä¿å­˜ã•ã‚Œãªã„ãŸã‚ã€åˆ¥ã®ã‚ªãƒ–ã‚¸ã‚§ã‚¯ãƒˆã«æ ¼ç´ã™ã‚‹å¿…è¦ãŒã‚ã‚‹

```R
df1_sub <- filter(df1, Tum1 > 100)

# ä¸Šã¨åŒã˜æ„å‘³
filter(df1, Tum1 > 100) -> df1_sub
```

## 7. ç‰¹å®šã®åˆ—ã‚’æŠ½å‡ºã™ã‚‹ï¼š `select()`

- `select()` ã¯ç¬¬ï¼‘å¼•æ•°ã«ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ ã€ç¬¬ï¼’ï¼ˆä»¥é™ï¼‰å¼•æ•°ã«åˆ—åã‚’æ›¸ãã¨ã€æ¡ä»¶ã«å½“ã¦ã¯ã¾ã‚‹åˆ—ã‚’æŠ½å‡ºã™ã‚‹
  - åˆ—åã«ã‚¹ãƒšãƒ¼ã‚¹ãŒå…¥ã£ã¦ã„ã‚‹å ´åˆã¯ãƒãƒƒã‚¯ã‚¹ãƒ©ãƒƒã‚·ãƒ¥ï¼ˆ ` ï¼‰ã§å›²ã‚€å¿…è¦ãŒã‚ã‚‹

ä»¥ä¸‹ã«ä¾‹ã‚’ç¤ºã™

```R
# mice_gene_symbol åˆ—ã‚’æŠ½å‡ºã™ã‚‹
select(df1, mice_gene_symbol)

# mice_gene_symbol åˆ—ã¨WT1åˆ—ã‚’æŠ½å‡ºã™ã‚‹
select(df1, mice_gene_symbol, WT1)

# mice_gene_symbol åˆ—ã‚’æŠ½å‡ºã™ã‚‹
select(df1, mice_gene_symbol, human_gene_symbol, WT1)

# mice_gene_symbol åˆ—ä»¥å¤–ã‚’æŠ½å‡ºã™ã‚‹
# (-ã‚’å‰ã«ã¤ã‘ã‚‹ã¨ãã®åˆ—ã‚’é™¤ã)
select(df1, -mice_gene_symbol)
```

ãªãŠã€æŠ½å‡ºã•ã‚ŒãŸçµæœã¯ã€å…ƒã®ãƒ‡ãƒ¼ã‚¿ãƒ•ãƒ¬ãƒ¼ãƒ è‡ªä½“ã«ã¯ä¿å­˜ã•ã‚Œãªã„ãŸã‚ã€åˆ¥ã®ã‚ªãƒ–ã‚¸ã‚§ã‚¯ãƒˆã«æ ¼ç´ã™ã‚‹å¿…è¦ãŒã‚ã‚‹

```R
df1_selected <- select(df1, mice_gene_symbol, WT1)

# ä¸Šã¨åŒã˜æ„å‘³
select(df1, mice_gene_symbol, WT1) -> df1_selected
```

## 8. ãƒ‘ã‚¤ãƒ—æ¼”ç®—å­ `%>%`

- `%>%` â†’ ã“ã‚Œã®ã“ã¨
- `tidyverse` ãƒ‘ãƒƒã‚±ãƒ¼ã‚¸ã§ä½¿ç”¨ã§ãã‚‹

ä»¥ä¸‹ã«ä¾‹ã‚’ç¤ºã™

```R
# str(df1) ã¨åŒã˜
df1 %>% str
```

```R
# select(df1, mice_gene_symbol, WT1) ã¨åŒã˜
df1 %>% select(mice_gene_symbol, WT1)

# %>% ã®å¾Œã«æ”¹è¡Œã‚„ã‚¤ãƒ³ãƒ‡ãƒ³ãƒˆãŒå…¥ã£ã¦ã‚‚åŒã˜æ„å‘³
df1 %>% 
    select(mice_gene_symbol, WT1)
```

```R
# select(filter(df1, Tum1>100), WT) ã¨åŒã˜æ„å‘³
# (ãƒ‘ã‚¤ãƒ—ã‚’ã¤ãªã’ã‚‹ã“ã¨ã‚‚ã§ãã‚‹)
df1 %>% 
    filter(Tum1>100) %>%
    select(mice_gene_symbol, WT1)

# çµæœã‚’æ ¼ç´ã™ã‚‹ã“ã¨ã‚‚ã§ãã‚‹
df1 %>% 
    filter(Tum1>100) %>%
    select(mice_gene_symbol, WT1) -> df1_filtered_selected
```

## 9. åˆ—ã‚’åŠ ãˆã‚‹ï¼š `mutate()`

- `mutate()` ã¯æ–°ã—ã„åˆ—ã‚’åŠ ãˆã‚‹ã“ã¨ãŒã§ãã‚‹
  - ç¬¬ï¼’å¼•æ•°ã§ã¯ã€`=` ã®å‰ã«æ–°ã—ã„åˆ—åã‚’æ›¸ãã€`=` ã®å¾Œã«æ¡ä»¶ã‚’æ›¸ã

```R
mutate(df1, Tum123_mean = (Tum1+Tum2+Tum3)/3)

# ä¸Šã¨åŒã˜
df1 %>% mutate(Tum123_mean = (Tum1+Tum2+Tum3)/3)
```

```R
# åŒæ™‚ã«è¤‡æ•°ã‚‚å¯èƒ½
mutate(df1, Tum123_mean = (Tum1+Tum2+Tum3)/3, WT12_mean = (WT1+WT2)/2)

# ä¸Šã¨åŒã˜
df1 %>% mutate(Tum123_mean = (Tum1+Tum2+Tum3)/3, WT12_mean = (WT1+WT2)/2)
```

## 8. ï¼’ã¤ã®åˆ—ã‚’çµåˆã™ã‚‹

- ï¼’ã¤ã®åˆ—ã‚’çµåˆã™ã‚‹æ“ä½œã‚’ä¸€èˆ¬ã« JOIN (ã˜ã‚‡ã„ã‚“)ã¨å‘¼ã¶
- ï¼’ã¤ã®è¡¨ã‚’çµåˆã™ã‚‹ãŸã‚ã®åˆ—ã‚’ã‚­ãƒ¼ï¼ˆkeyï¼‰ã¨å‘¼ã¶



![](img/2021-02-07-00-39-59.png)

![](img/2021-02-07-00-40-45.png)

```R
df2 <- read_tsv("data/HOM_MouseHumanSequence.rpt", col_names=TRUE)

# (ã‚­ãƒ¼ã®åå‰ãŒï¼’ã¤ã®è¡¨ã§é•ã†æ™‚ã¯ä»¥ä¸‹ã®ã‚ˆã†ã«æ›¸ãã¨ã‚ˆã„)
inner_join(df1, df2, by = c("mice_gene_symbol"="Symbol"))

# ä¸Šã¨åŒã˜
df1 %>% 
    inner_join(df2, by = c("mice_gene_symbol"="Symbol"))

# join ã®çµæœã‚’æ–°ãŸãªã‚ªãƒ–ã‚¸ã‚§ã‚¯ãƒˆã«æ ¼ç´ã™ã‚‹
df1 %>% 
    inner_join(df2, by = c("mice_gene_symbol"="Symbol")) -> df1_and_2
```

## `tumor_communication.ipynb` ã‚’å®Ÿè¡Œã™ã‚‹

### ã“ã‚Œã¯ãªã‚“ãªã®ã‹ï¼Ÿ

- ã€Œéå°ç´°èƒè‚ºç™Œ (non-small-cell lung cancer; NSCLC)ã«ãŠã„ã¦ã€è…«ç˜çµ„ç¹”å†…ã«å­˜åœ¨ã™ã‚‹é–“è³ªç´°èƒã‹ã‚‰è…«ç˜ç´°èƒã«ã‚·ã‚°ãƒŠãƒ«ã®ã‚¯ãƒ­ã‚¹ãƒˆãƒ¼ã‚¯ãŒèµ·ã“ã£ã¦è…«ç˜ãŒæ´»æ€§åŒ–ã•ã‚Œã‚‹ã‹ï¼Ÿ ã€ã‚’èª¿ã¹ã¦ã¿ã‚‹è§£æã‚’Rã§è¡Œã†ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯
  - `.ipynb` ã¨ã„ã†ã®ãŒãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã®æ‹¡å¼µå­ã§ã™
- ãƒ‡ãƒ¼ã‚¿ã®å…ƒè«–æ–‡
  - Toi et al., Transcriptome Analysis of Individual Stromal Cell Populations Identifies Stroma-Tumor Crosstalk in Mouse Lung Cancer Model, Cell Reports (2015)
  - https://doi.org/10.1016/j.celrep.2015.01.040
- ã‚µãƒ³ãƒ—ãƒ«ãƒ»è¨ˆæ¸¬æŠ€è¡“ã«ã¤ã„ã¦
  - è…«ç˜ãƒ¢ãƒ‡ãƒ«ãƒã‚¦ã‚¹ç”±æ¥ã®RNA-seqãƒ‡ãƒ¼ã‚¿
  - ï¼•ç¨®é¡ã®çµ„ç¹”ãƒ»ç´°èƒç¾¤ x 2ç¨®é¡ã®æ¡ä»¶ï¼š
    - ï¼•ç¨®é¡ã®çµ„ç¹”ãƒ»ç´°èƒç¾¤ç´°èƒç¾¤ï¼š
      - NSCLCãƒ¢ãƒ‡ãƒ«ãƒã‚¦ã‚¹ã¨é‡ç”Ÿå‹ã®ãƒã‚¦ã‚¹ã®è‚ºã‹ã‚‰ã‚»ãƒ«ã‚½ãƒ¼ã‚¿ãƒ¼ã§åˆ†ã‘ãŸã‚‚ã®ï¼šãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã€å˜çƒç´°èƒã€å¥½ä¸­çƒã€ä¸Šçš®ç´°èƒ
        - ç´°èƒå‹ã®åˆ†é¡ã¯ç´°èƒè¡¨é¢ã®æŠ—åŸãƒãƒ¼ã‚«ãƒ¼ã«ã‚ˆã‚‹
          - <img src="img/2021-02-07-09-08-20.png" alt="" width="200">
      - Total lung cells
    - 2ç¨®é¡ã®æ¡ä»¶: WT ï¼ˆé‡ç”Ÿå‹; Wild type)ã€Tum (è…«ç˜; Tumor)
- è§£æã®æ–¹é‡
  - é–“è³ªç´°èƒã«ãŠã„ã¦ã€Œæ­£å¸¸çµ„ç¹”ç”±æ¥ã®ã‚µãƒ³ãƒ—ãƒ«ã€ã«æ¯”ã¹ã¦ã€Œè…«ç˜çµ„ç¹”ç”±æ¥ã®ã‚µãƒ³ãƒ—ãƒ«ã€ã§ç™ºç¾é‡ãŒå¢—åŠ ã—ã¦ã„ã‚‹éºä¼å­ç¾¤ã‚’æŠ½å‡ºã™ã‚‹
  - è…«ç˜çµ„ç¹”ã«ãŠã„ã¦æ­£å¸¸çµ„ç¹”ã«ãŠã„ã¦ç™ºç¾é‡ãŒå¢—åŠ ã—ã¦ã„ã‚‹éºä¼å­ç¾¤ã‚’æŠ½å‡ºã™ã‚‹
  - ã€Œ1.ã«å«ã¾ã‚Œã‚‹ãƒªã‚¬ãƒ³ãƒ‰ã€ã¨ã€Œ2.ã«å«ã¾ã‚Œã‚‹å—å®¹ä½“ï¼ˆãƒ¬ã‚»ãƒ—ã‚¿ãƒ¼ï¼‰ã€ã‚’ãƒªã‚¬ãƒ³ãƒ‰ãƒ»å—å®¹ä½“ã®ãƒšã‚¢ã®ãƒ‡ãƒ¼ã‚¿ãƒ™ãƒ¼ã‚¹ã¨æ¯”è¼ƒã™ã‚‹


### å®Ÿè¡Œæ–¹æ³•

- [tumor_communication.ipynb](tumor_communication.ipynb) ã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã—ã¦ãã ã•ã„
- JupyterHub ã® `Upload` ã¨ã„ã†ãƒœã‚¿ãƒ³ã§ã‚¢ãƒƒãƒ—ãƒ­ãƒ¼ãƒ‰ã§ãã¾ã™
- ã‚¢ãƒƒãƒ—ãƒ­ãƒ¼ãƒ‰ã—ãŸ `.ipynb` å½¢å¼ã®ãƒ•ã‚¡ã‚¤ãƒ«ã¯ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã¨ã—ã¦é–‹ã„ã¦å®Ÿè¡Œãƒ»ç·¨é›†ãŒã§ãã¾ã™

### ä½¿ã†ãƒ‡ãƒ¼ã‚¿ã®æ§‹é€ 

- `gene_exp.diff`
  - ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã«ãŠã„ã¦ã€å„éºä¼å­ã®ç™ºç¾ã«WTã¨Tumã®æ¡ä»¶é–“ã§å·®ãŒã‚ã‚‹ã‹ã‚’èª¿ã¹ã‚‹ç™ºç¾å¤‰å‹•è§£æã®çµæœ
  - åˆ—ã®èª¬æ˜
    - test_id: ç•¥
    - gene_id: éºä¼å­IDï¼ˆã“ã“ã§ã¯éºä¼å­åãŒç”¨ã„ã‚‰ã‚Œã¦ã„ã‚‹ï¼‰
    - gene: ç•¥
    - locus: ç•¥
    - sample_1: ç•¥
    - sample_2: ç•¥
    - status: ç•¥
    - value_1: æ­£å¸¸çµ„ç¹”ã§ã®ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã«ãŠã‘ã‚‹éºä¼å­ç™ºç¾é‡ã®å¹³å‡å€¤
    - value_2: è…«ç˜çµ„ç¹”å†…ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ (intratumoral macropahges)ã§ã®éºä¼å­ç™ºç¾é‡ã®å¹³å‡å€¤
    - log2(fold_change): value_2/value_1ã‚’log2å¤‰æ›ã—ãŸã‚‚ã®ã€‚æ­£ã®å€¤ã§ã‚ã‚Œã° value_2 > value_1ã€è² ã®å€¤ã§ã‚ã‚Œã° value_1 > value_2
    - test_stat: ç•¥
    - p_value: ç™ºç¾å¤‰å‹•ã‚’åˆ¤å®šã™ã‚‹çµ±è¨ˆæ¤œå®šã®çµæœã®p-valueã€‚å°ã•ã„æ–¹ãŒçµ±è¨ˆçš„ã«æœ‰æ„ã€‚
    - q_value: p_value ã‚’å¤šé‡æ¤œå®šè£œæ­£ã—ãŸã‚‚ã®ã€‚å°ã•ã„æ–¹ãŒçµ±è¨ˆçš„ã«æœ‰æ„ã€‚
    - significant: ç•¥

ç¸¦ã«éºä¼å­åã®ã‚ˆã†ãªã‚‚ã®ãŒä¸¦ã‚“ã§ã„ã‚‹ã€‚å„è¡ŒãŒéºä¼å­ã€å„åˆ—ãŒå¤‰æ•°ã«ãªã£ã¦ã„ã‚‹
p_valueã‚„q_valueã¯ç™ºç¾å¤‰å‹•ã‚’åˆ¤å®šã™ã‚‹çµ±è¨ˆæ¤œå®šã®çµæœã ã‚ã†
log2(fold_change) ã¯ value_2/value_1ã‚’log2å¤‰æ›ã—ãŸã‚‚ã®ã ã‚ã†
ã¡ãªã¿ã«ã€value_1 ã¯æ­£å¸¸çµ„ç¹”ã§ã®ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã€ value_2 ã¯è…«ç˜çµ„ç¹”å†…ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ (intratumoral macropahges)ã§ã®FPKMã®å¹³å‡å€¤
ã“ã‚Œã¯ ãƒ¡ã‚¿ãƒ‡ãƒ¼ã‚¿ ã‚’ã¿ãªã„ã¨ã‚ã‹ã‚‰ãªã„

- `HOM_MouseHumanSequence.rpt`
  - ãƒ’ãƒˆ-ãƒã‚¦ã‚¹ã®ã‚ªãƒ¼ã‚½ãƒ­ã‚°é–¢ä¿‚ã®è¡¨ã€‚ç”Ÿç‰©ç¨®ã”ã¨ã®éºä¼å­ã«ã¤ã„ã¦è¨˜è¿°ã•ã‚Œã¦ã„ã‚‹ã€‚
  - åˆ—ã®èª¬æ˜
    - HomoloGene ID: ã‚ªãƒ¼ã‚½ãƒ­ã‚°é–¢ä¿‚è‡ªä½“ã®IDã€‚ã“ã‚ŒãŒäº’ã„ã«å…±é€šãªãƒ’ãƒˆéºä¼å­ãƒ»ãƒã‚¦ã‚¹éºä¼å­ã¯ã‚ªãƒ¼ã‚½ãƒ­ã‚°é–¢ä¿‚ã«ã‚ã‚‹
    - Common Organism Name: ç”Ÿç‰©ç¨®å
    - NCBI Taxon ID: ç•¥
    - Symbol: éºä¼å­ã‚·ãƒ³ãƒœãƒ«
    - EntrezGene ID
    - Mouse MGI ID: ç•¥
    - HGNC ID: ç•¥
    - OMIM Gene ID: ç•¥
    - Genetic Location: ç•¥
    - Genomic Coordinates (mouse: GRCm38, human: GRCh37.p10): ç•¥
    - Nucleotide RefSeq IDs: ç•¥
    - Protein RefSeq IDs: ç•¥
    - SWISS_PROT IDs: ç•¥

- `genes.read_group_tracking`
  - è…«ç˜ç´°èƒã«ãŠã„ã¦å„éºä¼å­ã®ç™ºç¾é‡ã‚’ã¾ã¨ã‚ãŸè¡¨
  - åˆ—ã®èª¬æ˜
    - tracking_id: éºä¼å­ã®ID
    - condition: ç•¥
    - replicate: ç•¥
    - raw_frags: ç•¥
    - internal_scaled_frags: ç•¥
    - external_scaled_frags: ç•¥
    - FPKM: æ­£è¦åŒ–ã•ã‚ŒãŸéºä¼å­ç™ºç¾é‡ã€‚å¤§ãã„æ–¹ãŒéºä¼å­ç™ºç¾é‡ï¼ˆmRNAé‡ï¼‰ãŒå¤šã„ã€‚
    - effective_length: ç•¥
    - status: ç•¥

- `LR_manual_revised.txt`
  - ãƒ’ãƒˆã«ãŠã‘ã‚‹ã€ãƒªã‚¬ãƒ³ãƒ‰ã¨å—å®¹ä½“ã®ãƒšã‚¢ã®å¯¾å¿œé–¢ä¿‚ã‚’ã¾ã¨ã‚ãŸè¡¨
  - åˆ—ã®èª¬æ˜
    - From: ãƒªã‚¬ãƒ³ãƒ‰ã®éºä¼å­ã‚·ãƒ³ãƒœãƒ«
    - To: å—å®¹ä½“ã®éºä¼å­ã‚·ãƒ³ãƒœãƒ«  


- `GSE59831_processed_data_FPKM.txt`
  - å„ã‚µãƒ³ãƒ—ãƒ«ã§ã®å„éºä¼å­ã®ç™ºç¾é‡ã‚’èª¿ã¹ãŸè¡¨
    - Tum1, Tum2, Tum3 ãŒè…«ç˜çµ„ç¹”å†…ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã€WT1, WT2ãŒé‡ç”Ÿå‹çµ„ç¹”ã§ã®ãƒã‚¯ãƒ­ãƒ•ã‚¡ãƒ¼ã‚¸ã€Tum9, Tum10, Tum11ãŒè…«ç˜ç´°èƒã«ãŠã‘ã‚‹éºä¼å­ç™ºç¾é‡
  - åˆ—ã®èª¬æ˜
    - mice_gene_symbol: ãƒã‚¦ã‚¹ã®ãŠã‘ã‚‹éºä¼å­ã‚·ãƒ³ãƒœãƒ«
    - human_gene_symbol: ãƒ’ãƒˆã®ãŠã‘ã‚‹éºä¼å­ã‚·ãƒ³ãƒœãƒ«
    - Tum1
    - Tum2
    - Tum3
    - WT1
    - WT2
    - Tum4
    - Tum5
    - WT3
    - WT4
    - Tum6
    - Tum7
    - Tum8
    - WT5
    - WT6
    - WT7
    - Tum9
    - Tum10
    - Tum11
    - WT8
    - WT9
    - WT10
    - Tum12
    - Tum13
    - Tum14
    - WT11
    - WT12
    - WT13



## åŸºæœ¬èª²é¡Œ

### åŸºæœ¬èª²é¡ŒA-1

- ä»¥ä¸‹ã®ã‚ˆã†ãªRã®ã‚³ãƒ¼ãƒ‰ã‚’ä½œæˆã—ãªã•ã„
  - `tidyverse` ã‚’ä½¿ã†
  - `HOM_MouseHumanSequence.rpt` ã‚’èª­ã¿è¾¼ã‚“ã§ã€ `df_hom` ã¨ã„ã†Rã‚ªãƒ–ã‚¸ã‚§ã‚¯ãƒˆã«æ ¼ç´ã™ã‚‹
  - è¡Œæ•°ã¨åˆ—æ•°ã‚’èª¿ã¹ã‚‹
- ä¸Šã®ãƒ¡ãƒ‹ãƒ¥ãƒ¼ã‹ã‚‰ `File > Download as > R (.r)` ã¨ã™ã‚‹ã“ã¨ã§ã€Rã®ã‚³ãƒ¼ãƒ‰ã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã§ãã‚‹ã®ã§ã€ãã‚Œã‚’ manaba ã§æå‡ºã›ã‚ˆ

### åŸºæœ¬èª²é¡ŒA-2

- ä»¥ä¸‹ã®ã‚ˆã†ãªã‚³ãƒ¼ãƒ‰ã‚’ä½œæˆã—ãªã•ã„
  - `dim(df1)` ã‚’ãƒ‘ã‚¤ãƒ—æ¼”ç®—å­ã‚’ä½¿ã£ã¦æ›¸ãæ›ãˆã‚‹
- manaba ã«ç›´æ¥æ›¸ãè¾¼ã‚

### åŸºæœ¬èª²é¡ŒA-3

- df1 ã¨ df2 ã‚’ `inner_join()`, `left_join()`, `right_join()`, `outer_join()` ã§JOINã—ãŸéš›ã«ã€ãã‚Œãã‚Œã®çµæœã®è¡Œæ•°ã‚’èª¿ã¹ã‚ˆ
  - ãŸã ã—ã€ã‚­ãƒ¼ã¯ c("human_gene_symbol"="Symbol")
- manaba ã«ç›´æ¥æ›¸ãè¾¼ã‚
  - åŠè§’ã§ï¼”ç¨®é¡ã®è¡Œæ•°ã‚’ã‚³ãƒ³ãƒï¼ˆ`,`ï¼‰ã§åŒºåˆ‡ã‚Œï¼ˆä¾‹ï¼š `103,130,142,210`ï¼‰

### åŸºæœ¬èª²é¡ŒA-4

- `LR_manual_revised.txt` ã‚’èª­ã¿è¾¼ã¿ã€ `From`åˆ—ã¨`To`åˆ—ã«ãã‚Œãã‚Œãƒ¦ãƒ‹ãƒ¼ã‚¯ãªè¦ç´ ãŒã„ãã¤ã‚ã‚‹ã‹ã‚’èª¿ã¹ã‚‹Rã®ã‚³ãƒ¼ãƒ‰ã‚’ä½œæˆã—ã€å®Ÿè¡Œã›ã‚ˆ
  - ãƒ’ãƒ³ãƒˆï¼šunique() ã‚’ä½¿ã†ã¨ã€é‡è¤‡ã®ã‚ã‚‹è¡Œã‚’é™¤ãã“ã¨ãŒã§ãã‚‹
- æ–°ã—ã„ãƒãƒ¼ãƒˆãƒ–ãƒƒã‚¯ã‚’ä½œæˆã—ã¦ä½œæˆã™ã‚‹ã“ã¨
- ä¸Šã®ãƒ¡ãƒ‹ãƒ¥ãƒ¼ã‹ã‚‰ `File > Download as > PDF (.pdf)` ã¨ã™ã‚‹ã“ã¨ã§ã€Rã®ã‚³ãƒ¼ãƒ‰ã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã§ãã‚‹ã®ã§ã€ãã‚Œã‚’ manaba ã§æå‡ºã›ã‚ˆ

### åŸºæœ¬èª²é¡ŒA-5

- [tumor_communication.ipynb](tumor_communication.ipynb) ã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã—ã€å®Ÿè¡Œã›ã‚ˆ
- ä¸Šã®ãƒ¡ãƒ‹ãƒ¥ãƒ¼ã‹ã‚‰ `File > Download as > PDF (.pdf)` ã¨ã™ã‚‹ã“ã¨ã§ã€å®Ÿè¡Œçµæœã‚’ãƒ€ã‚¦ãƒ³ãƒ­ãƒ¼ãƒ‰ã§ãã‚‹ã®ã§ã€ãã‚Œã‚’ manaba ã§æå‡ºã›ã‚ˆ
